src/backend/marsha/core/migrations/0016_auto_20191001_0812.py

```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.2.5 on 2019-10-01 08:12
+
+import json
+import re
+
+from django.conf import settings
+from django.db import migrations
+
+import boto3
+
+
+def escape_timed_text_tracks(apps, schema_editor):
+    if not settings.SHOULD_RUN_TIMEDTEXT_TRACKS_ESCAPING_MIGRATION:
+        return
+
+    marsha_s3 = boto3.client(
+        "s3",
+        region_name=settings.AWS_S3_REGION_NAME,
+        aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+    )
+    marsha_lambda = boto3.client(
+        "lambda",
+        region_name=settings.AWS_S3_REGION_NAME,
+        aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+    )
+    base_aws_name = settings.AWS_SOURCE_BUCKET_NAME.rstrip("-source")
+    lambda_function_name = f"{base_aws_name}-encode"
+    paginator = marsha_s3.get_paginator("list_objects")
+    response_itrator = paginator.paginate(Bucket=settings.AWS_SOURCE_BUCKET_NAME)
+
+    regex = re.compile(r"^.*\/timedtexttrack\/.*$")
+
+    for response in response_itrator:
+        for ttt in response.get("Contents"):
+            if regex.match(ttt.get("Key")):
+                marsha_lambda.invoke(
+                    InvocationType="Event",
+                    FunctionName=lambda_function_name,
+                    Payload=json.dumps(
+                        {
+                            "Records": [
+                                {
+                                    "s3": {
+                                        "object": {"key": ttt.get("Key")},
+                                        "bucket": {
+                                            "name": settings.AWS_SOURCE_BUCKET_NAME
+                                        },
+                                    }
+                                }
+                            ]
+                        }
+                    ),
+                )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [("core", "0015_auto_20190904_1234")]
+
+    operations = [migrations.RunPython(escape_timed_text_tracks)]
```

Comment:
are you sure you can skip indicating the method for reverse migration? Try running the reverse migration to be sure?

Reply:
Good catch. without reverse code the migration can't be undo.